src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRunner.cs

```diff
@@ -23,9 +24,9 @@ public class ParameterIdentificationRunner : IParameterIdentificationRunner
       private readonly IDialogCreator _dialogCreator;
       private readonly IEntityValidationTask _entityValidationTask;
       private readonly IOSPSuiteExecutionContext _executionContext;
-      private IParameterIdentificationEngine _parameterIdentificationEngine;
+      private Cache<string, IParameterIdentificationEngine> _parameterIdentificationEngines;
```

Comment:
I am not sure we will always have the same object for a PI I guess if at some point we clone or something but still keep the same ID (meaning it refers to the same PI) then we should identify the PI by its Id and not by the object itself. This is why I used Id and not the object.

Reply:
There should never be two objects with the same Id. It will break everything. 
Id are only used for
 - Serialization
 - Command so that we can retrieve an object by id without holding a reference 

Otherwise we always use reference. That's why I insist on using reference instead of id, especially to retrieve the same instance later on by id

Sometimes we store reference with "path" but this is for faster retrieval and this makes sense